[KibanaPageLayout] Solution Nav specific styles & props (#100089)
elastic/kibana: dev_docs/tutorials/kibana_page_template.mdx

## `solutionNav`

To add left side navigation for your solution, we recommend passing [**EuiSideNav**](https://elastic.github.io/eui/#/navigation/side-nav) props to the `solutionNav` prop. The template component will then handle the mobile views and add the solution nav embellishments. On top of the EUI props, you'll need to pass your solution `name` and an optional `icon`.

If you need to custom side bar content, you will need to pass you own navigation component to `pageSideBar`. We still recommend using [**EuiSideNav**](https://elastic.github.io/eui/#/navigation/side-nav).

When using `EuiSideNav`, root level items should not be linked but provide section labelling only.

```tsx
<KibanaPageLayout
  solutionNav={{
    name: 'Management',
    icon: 'managementApp',
    items: [
      {
        name: 'Root item',
        items: [
          { name: 'Navigation item', href: '#' },
          { name: 'Navigation item', href: '#' },
        ]
      }
    ]
  }}
>
  {...}
</KibanaPageLayout>
```

> ![Screenshot of Stack Management empty state with a provided solution navigation.](../assets/kibana_template_solution_nav.png)

Reviewer: Adding some more notes describing the image that might be useful to low-vision users to piece together the image. 

```suggestion
![Screenshot of Stack Management empty state with a provided solution navigation shown on the left, outlined in pink.](../assets/kibana_template_solution_nav.png)
```

> ![Screenshots of Stack Management page in mobile view. One with the menu closed, one with it open.](../assets/kibana_template_solution_nav_mobile.png)

Reviewer: ```suggestion
![Screenshots of Stack Management page in mobile view. Menu closed on the left, menu open on the right.](../assets/kibana_template_solution_nav_mobile.png)
```
